### ISSUE TITLE
Prefix Database Tables

### ISSUE THREAD

**Issue by jlarmstrongiv** [prisma/prisma#2278]
## Problem
Unfortunately, I am working on brownfield projects that share a database.

## Solution
It would be nice to be able to set an .env variable to prefix all the tables that prisma will use.

## Alternatives
Prefix all of them in the `schema.prisma` myself

## Additional context
https://github.com/prisma/prisma/discussions/2232 Keep up the good work 😄

**Comment by benatkin** [comment]
Does the alternative work with prisma migrate?

```
Prefix all of them in the schema.prisma myself
```
 
Prefixing tables is very common functionality outside of enterprise as well. WordPress prefixes everything with `wp_` and lets you change the prefix.

**Comment by janpio** [comment]
> Does the alternative work with prisma migrate?

Yes, Prisma Migrate uses the model/table names from the schema file.

**Comment by benatkin** [comment]
What I mean is, could there be two Prisma apps in one database, or would they both try to use the same migrations table? IIRC it's called `_migrations`

**Comment by pantharshit00** [comment]
You can use the client but migrate won't work as it will try to delete the extra tables from the other project. Maybe you can workaround that by using `@@ignore` but I will not recommend that.

**Comment by aniruddha-adhikary** [comment]
We had to implement a `sed` replacement in the build/CI process. Extremely hacky. Hope we can get table prefix support soon.

**Comment by janpio** [comment]
Where would you expect to configure this setting?
How would you expect Prisma Client to handle the model names?
Would you expect these models being able to interact with non prefixed tables (e.g. via relations)?

**Comment by jlarmstrongiv** [comment]
@janpio 

> Where would you expect to configure this setting?

I would expect to configure this setting in the datasource, and using environment variables `prefix = env("PREFIX")` is a must.  No other changes to the schema required.

> How would you expect Prisma Client to handle the model names?

Non-prefixed.  My code would remain the same `const user = await prisma.user.create({ … })`

> Would you expect these models being able to interact with non prefixed tables (e.g. via relations)?

Yes, but only with specific non-prefixed tables defined by introspection or myself using the `@@map`.  I may be missing the intention of your last question, let me know if you have followups.

**Comment by janpio** [comment]
Similar suggestion, but for the model names in the Prisma schema: https://github.com/prisma/prisma/issues/9816

**Comment by GainorB** [comment]
Update?

**Comment by yasz** [comment]
any update?

**Comment by westtrade** [comment]
Updates?

**Comment by itintouchmk** [comment]
Updates?
